sormas-ui/src/main/java/de/symeda/sormas/ui/dashboard/campaigns/CampaignDashboardDiagramComponent.java

```diff
@@ -66,23 +65,29 @@ public CampaignDashboardDiagramComponent(
 		setMargin(false);
 		addComponent(campaignColumnChart);
 
+		final Map<Object, String> axisInfo = new HashMap<>();
 		for (CampaignDiagramDataDto diagramData : diagramDataList) {
-			if (!axisKeys.contains(diagramData.getGroupingKey())) {
-				axisKeys.add(diagramData.getGroupingKey());
-				axisCaptions.put(diagramData.getGroupingKey(), diagramData.getGroupingCaption());
+			final Object groupingKey = diagramData.getGroupingKey();
+			if (!axisInfo.containsKey(groupingKey)) {
+				axisInfo.put(groupingKey, diagramData.getGroupingCaption());
 			}
 
 			String seriesKey = diagramData.getFormId() + diagramData.getFieldId();
 			if (!diagramDataBySeriesAndXAxis.containsKey(seriesKey)) {
 				diagramDataBySeriesAndXAxis.put(seriesKey, new HashMap<>());
 			}
 			Map<Object, CampaignDiagramDataDto> objectCampaignDiagramDataDtoMap = diagramDataBySeriesAndXAxis.get(seriesKey);
-			if (objectCampaignDiagramDataDtoMap.containsKey(diagramData.getGroupingKey())) {
+			if (objectCampaignDiagramDataDtoMap.containsKey(groupingKey)) {
 				throw new RuntimeException("Campaign diagram data map already contains grouping");
 			}
-			objectCampaignDiagramDataDtoMap.put(diagramData.getGroupingKey(), diagramData);
+			objectCampaignDiagramDataDtoMap.put(groupingKey, diagramData);
 		}
 
+		xAxisInfo = axisInfo.entrySet()
+			.stream()
+			.sorted(Map.Entry.comparingByValue())
```

Comment:
with my example data i found it would be nice to have a case insensitive sorting here (that is a should be before W)